### ISSUE TITLE
Collecting Requirements

### ISSUE THREAD

**Issue by oldcai** [oldcai/ITPMAssignment#3]
Hi, since we have chosen the loyalty plan, we can start to collect requirements now.

GitHub is a better place than chatbox since the topics are separated in a organized way.

**Comment by Nagasubramaniyan** [comment]
Business Requirements:

The customer logs into the loyalty program web application to check for the offer information and redeems. 
There can be a flat-file system or a Database management system to handle the customer information and offer information for each customer respectively.

There will be two categories of offers listed to the customer one is general (such as first time user, Credit card offers, Cashbacks,etc.,) and another one user-specific (such as special discounts, wallet offers, etc.,) displayed based on the frequency factor which is the key filter for providing the offers. 

The frequency factor can be determined by the organization as the customer ticket booking and travel criteria (more than 5 ticket bookings then for 3 more tickets, the offer may be updated and also based on the membership (silver, gold, platinum)). Once the coupon is selected by the customer for any ticket booking then the offer is updated as used and not valid anymore in the data source.

Reference links: 
https://www.openloyalty.io/
https://antavo.com/blog/customer-loyalty-programs
https://www.voucherify.io/blog/architecture-of-customer-loyalty-software-a-guide-for-product-managers
                     
Technology recommended:
Front end: Bootstrap, Javascript, and jQuery
Back end: Spring Boot 
Database: H2

Reference link: https://spring.io/projects/spring-boot

**Comment by oldcai** [comment]
> Business Requirements:
> 
> The customer logs into the loyalty program web application to check for the offer information and redeems.
> There can be a flat-file system or a Database management system to handle the customer information and offer information for each customer respectively.
> 
> There will be two categories of offers listed to the customer one is general (such as first time user, Credit card offers, Cashbacks,etc.,) and another one user-specific (such as special discounts, wallet offers, etc.,) displayed based on the frequency factor which is the key filter for providing the offers.
> 
> The frequency factor can be determined by the organization as the customer ticket booking and travel criteria (more than 5 ticket bookings then for 3 more tickets, the offer may be updated and also based on the membership (silver, gold, platinum)). Once the coupon is selected by the customer for any ticket booking then the offer is updated as used and not valid anymore in the data source.
> 
> Reference links:
> https://www.openloyalty.io/
> https://antavo.com/blog/customer-loyalty-programs
> https://www.voucherify.io/blog/architecture-of-customer-loyalty-software-a-guide-for-product-managers
> 
> Technology recommended:
> Front end: Bootstrap, Javascript, and jQuery
> Back end: Spring Boot
> Database: H2
> 
> Reference link: https://spring.io/projects/spring-boot

I agree with the front end part. But I recommend using Django as the backend solution since it needs less code to implement the same functions.

**Comment by oldcai** [comment]
![image](https://user-images.githubusercontent.com/1150130/96688016-3e495880-13b3-11eb-892d-ca745beb9ba1.png)
This might be helpful when implementing the system.
